libs/blocks/global-navigation/global-navigation.js

```diff
@@ -479,6 +482,35 @@ class Gnav {
     return decoratedElem;
   };
 
+  decorateAside = async () => {
+    this.elements.aside = '';
+    const promoPath = getMetadata('gnav-promo-source');
+    if (!promoPath) return this.elements.aside;
+
+    const onError = () => {
+      this.el.classList.remove('has-promo');
+      lanaLog({ message: 'Gnav Promo fragment not replaced, potential CLS' });
+      return this.elements.aside;
+    };
+
+    const fragLink = toFragment`<a href="${promoPath}">${promoPath}</a>`;
+    const fragTemplate = toFragment`<div>${fragLink}</div>`;
+    decorateAutoBlock(fragLink);
+    if (!fragLink.classList.contains('fragment')) return onError();
+    await loadMiloBlock(fragLink);
```

Comment:
We should add a catch either here, or even better around the whole block.

As our logic currently stands, if this promise doesn't resolve the page doesn't have a nav at all.

Reply:
Good idea, I'll see what makes sense to wrap so we avoid any issues 👍